portfolio/src/main/java/com/google/sps/servlets/models/Utils.java

```diff
@@ -0,0 +1,24 @@
+package com.google.sps.servlets.models;
+
+import java.util.List;
+import java.util.Map;
+import java.util.HashMap;
+import java.util.Arrays;
+import java.util.stream.Collectors;
+
+public abstract class Utils {
+  public static Map<String, String> extractFrom(String requestString, String[] params) {
+    List<String> dataList = Arrays.asList(requestString.split("\\n+"));
+    String initials = "Content-Disposition: form-data; ";
+
+    Map<String, String> results = Arrays.asList(params)
+      .stream()
+      .distinct()
+      .collect(Collectors.toMap(
+        s -> s, 
+        s -> dataList.get(dataList.indexOf(String.format("%sname=\"%s\"", initials, s)) + 1)
```

Comment:
Since DataForm is used to secure data transaction in the frontend, only elements with the attribute "name" specified are sent to the backend, thus "name=" will always be there when we extract the request. 

Source[url: https://stackoverflow.com/questions/12543848/does-form-data-still-transfer-if-the-input-tag-has-no-name]

Reply:
What if I make the request to your endpoint myself and skip over your form?  Never underestimate the ability of the users to screw with your system.  Your server should work independently of the client since it's always possible to interact with the server bypassing the client entirely.

Thanksgiving 2017 I spent several days investigating a bug that, at the end of the day, was someone using a custom job to make requests to our server (probably some kind of scraper).  After we made some changes, their request was malformed, which our code didn't expect, and the default error handling behavior was to redirect.  Their job would then, evidently, retry the same URL, and the end result of this feedback loop was a huge spike in client errors for that particular page.